glean-core/android/src/main/java/mozilla/telemetry/glean/private/BooleanMetricType.kt

```diff
@@ -14,16 +14,31 @@ import mozilla.telemetry.glean.rust.RustError
 import mozilla.telemetry.glean.Dispatchers
 import mozilla.telemetry.glean.rust.toBoolean
 
-class BooleanMetricType(
-    disabled: Boolean,
-    category: String,
-    lifetime: Lifetime,
-    name: String,
+/**
+ * This implements the developer facing API for recording boolean metrics.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time,
+ * allowing developers to record values that were previously registered in the metrics.yaml file.
+ *
+ * The boolean API only exposes the [set] method.
+ *
+ * The internal constructor is only used by labeled metrics directly.
```

Comment:
nit: you can change this to `"The internal constructor is only used by [LabeledMetricType] directly"` so that the generated docs point to that type.